nemo/core/classes/modelPT.py

```diff
@@ -152,91 +158,104 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
-                                    f"Could not find {used_src} inside "
-                                    f"tarfile {_MODEL_RESTORE_PATH} or under local"
-                                )
-                    finally:
-                        # change back working directory
-                        os.chdir(cwd)
-                else:
-                    # No further action can be taken, file not found anywhere
-                    raise FileNotFoundError(f"Could not find {used_src}")
-            else:
-                # Found filepath
-                archive_item.path = used_src
-
-            # But disregarding whether you use "local" or "remote" artifact - always store the original path.
-            # This fixes issues raising when finetuning NLP models that create and register tokenizer vocabs.
-            if config_path in self.artifacts:
-                logging.warning(
-                    f"Artifact {config_path} with value '{self.artifacts[config_path]}' "
-                    f"already exists and will be overwritten with value '{src}'!"
-                )
+        if config_path in self.artifacts.keys():
+            raise ValueError(
+                f"You tried to register an artifact under config key={config_path} but an artifact for"
+                f"it has already been registered."
+            )
 
-            self.artifacts[config_path] = archive_item
-            return used_src
+        # src is a local existing path - register artifact and return exact same path for usage by the model
+        if os.path.exists(os.path.abspath(src)):
+            return_path = os.path.abspath(src)
+        # this is the case when artifact must be retried from the nemo file
+        # we are assuming that the location of the right nemo file is available from _MODEL_RESTORE_PATH
+        elif src.startswith("nemo:"):
+            tmpdir = _CACHE_ROOT + f"_TMPDIRS/{uuid.uuid4().hex}"
+            os.makedirs(tmpdir)
+            if not hasattr(self, 'to_delete'):
+                self.to_delete = set()
+            self.to_delete.add(tmpdir)
+            outfolder = self._unpack_nemo_file(path2file=_MODEL_RESTORE_PATH, out_folder=tmpdir)
+            return_path = os.path.abspath(os.path.join(outfolder, src[5:]))
         else:
-            return src
+            raise FileNotFoundError(
+                f"src path does not exist or it is not a path in nemo file. src value I got was: {src}. Absolute: {os.path.abspath(src)}"
+            )
+
+        assert os.path.exists(return_path)
+        artifact_item = model_utils.ArtifactItem()
+        artifact_item.path = os.path.abspath(src)
+        artifact_item.path_type = model_utils.ArtifactPathType.LOCAL_PATH
+        self.artifacts[config_path] = artifact_item
+        # we were called by ModelPT
+        if hasattr(self, "_cfg"):
+            with open_dict(self._cfg):
+                self._cfg.update_node(config_path, return_path)
```

Comment:
Use self.cfg - so that wandb and loggers are also updated.

Reply:
done